tests/test_qasm_examples.py

```diff
@@ -69,58 +120,94 @@ def fix_ifs(line):
 
 
 def fix_spaces(line):
-    """Helper function to format spaces before/after paranthesis and after commas correctly"""
+    """Helper function to format spaces before/after paranthesis and after commas.
+
+    * `,` should never be surrounded by spaces.
+    * `(` should never be surrounded by spaces.
+    * `)` should never be preceeded by spaces, and should always
+      follow with one whitespace.
+    """
     line = ",".join(re.split(r"\s*,\s*", line))
     line = "(".join(re.split(r"\s*\(\s*", line))
     line = ") ".join(re.split(r"\s*\)\s*", line))
-    return clean(line)
+
+    return line.strip()
 
 
 # for variable names that are used by Python, and cannot be used as SymPy symbols
 invalid_var_names = ["lambda"]
 def fix_math_expr(line):
-    """Helper function to format math expressions correctly"""
+    """Helper function to format math expressions
 
-    # remove the if-parts from the string (doens't support maths in if-statments)
+    Checks for closed parantheses in a line from a qasm-file (ignoring
+    if-statements) and re-formats the mathematical expression(s) inside of it
+    using SymPy. For example, ``((alpha + beta) / 2)`` becomes ``(alpha/2 +
+    beta/2)``, and ``(4*pi/(1+3))`` becomes ``(pi)``.
```

Comment:
Are there never cases where math occurs not within matching parenthesis?

Reply:
I'm not absolutely sure, but it seems like there aren't. That's an assumption I've made here to make things a bit easier (although it's not fully bulletproof, and it might not be the best way of doing things if we want to expand what's allowed in the future either).